stellar/docs/link-token.md

```diff
@@ -0,0 +1,230 @@
+
+The token linking feature enables:
+
+- Linking existing tokens across connected Amplifier chains via ITS Hub
+- Supporting tokens with different decimal precisions through automatic scaling
+
+## How It Works
+
+The token linking process involves two key message types that ITS Hub uses to coordinate the linking:
+
+### Token Metadata Registration
+
+Before linking tokens, ITS Hub needs to know about each token's details (address and decimals) on both chains.
+
+### Token Linking
+
+Once metadata is registered, you can link tokens by specifying which tokens on different chains should be connected.
+
+### Decimal Scaling
+
+ITS Hub automatically handles decimal scaling between linked tokens:
+
+- **Stellar Token**: 7 decimals (1 USDC = 10,000,000 units)
+- **EVM Token**: 18 decimals (1 USDC = 1,000,000,000,000,000,000 units)
+- **Scaling Factor**: 10^(18-7) = 10^11
+
+When transferring 1 USDC from Stellar to EVM:
+
+- Stellar: 10,000,000 units locked/burned
+- EVM: 1,000,000,000,000,000,000 units unlocked/minted
+
+## Link Token Flow
+
+```
+1. User controls Token A (Stellar) and Token B (EVM)
+2. User → ITS Stellar: registerTokenMetadata(Token A) → ITS Hub
+3. User → ITS EVM: registerTokenMetadata(Token B) → ITS Hub
+4. User → ITS Stellar: registerCustomToken() → Deploys Token Manager A
+5. User → ITS Stellar: linkToken() → ITS Hub
+6. ITS Hub: Calculates scaling factor from stored decimals
+7. ITS Hub → ITS EVM: Deploy Token Manager B
+8. Token linking complete - InterchainTransfer enabled
+```
+
+## Prerequisites
+
+Before linking tokens, ensure you have:
+
+1. **Token Control**: You must control both tokens on their respective chains
+2. **Token Metadata**: Both tokens must be registered with their metadata
+3. **Gas Tokens**: Sufficient gas tokens for cross-chain operations
+4. **ITS Deployment**: ITS contracts must be deployed on both chains
+
+## Token Manager Types
+
+The following token manager types are supported:
+
+- `LOCK_UNLOCK` (2): For tokens that are locked on the source chain and unlocked on the destination chain
+- `MINT_BURN` (4): For tokens that are burned on the source chain and minted on the destination chain
+
+## Parameters
+
+**Required:**
+
+- `salt`: Unique identifier for the token linking operation
+- `tokenAddress`: Address of the token to be linked
+- `destinationChain`: Name of the destination chain
+- `destinationTokenAddress`: Address of the token on the destination chain
+- `type`: Token manager type (LOCK_UNLOCK, MINT_BURN)
+
+**Optional:**
+
+- `--operator`: Operator address for the token manager on the destination chain
+- `--gas-token-address`: Gas token address (defaults to XLM on Stellar)
+- `--gas-amount`: Gas amount for cross-chain operations
+
+## Step-by-Step Process
+
+### Step 1: Setup Tokens
+
+Setup Token with decimals on both chains:
+
+**Chain A (Stellar):**
+
+```bash
+ts-node stellar/its.js deploy-interchain-token <name> <symbol> <decimal> <salt> <initialSupply>
```

Comment:
These are test tokens. We are assuming the user already has existing tokens, so this needs to be labelled correctly